### PR TITLE
reprocess errors syncronously, add dryRun and env

### PR DESCRIPTION
**Pull request by eriktaubeneck**
this isn't the pretties JS but it should get the job done.

@tleach or @paetling could you review?

### CODE REVIEW

File: reprocessFailedBatches.js [hunk 1]
```diff
@@ -18,6 +19,14 @@ if (process.argv.length < 2) {
 }
 var setRegion = process.argv[2];
 
+if (process.argv.length < 3) {
+    console.log("You must prodive environment (prod, staging, all)");
```
Comment (paetling) [inline reprocessFailedBatches.js]: provide not 'prodive'

Comment (paetling) [inline reprocessFailedBatches.js]: also they could in theory provide the environment but not one of the other args, might just be worth printing out all arguments needed

Comment (eriktaubeneck) [inline reprocessFailedBatches.js]: fair point.

File: reprocessFailedBatches.js [hunk 2]
```diff
@@ -102,52 +111,57 @@ dynamoDB.query(queryItem, function(err, data) {
         process.exit(ERROR);
     } else {
         if (data && data.Items) {
+            async.eachSeries(data.Items, function(item, callback) {
+                validBatch = new RegExp('^' + 'gamechanger.insights.events.'+env).test(item['s3Prefix']['S'])
```
Comment (paetling) [inline reprocessFailedBatches.js]: why not just '^gamechanger...'

File: reprocessFailedBatches.js [hunk 3]
```diff
@@ -102,52 +111,57 @@ dynamoDB.query(queryItem, function(err, data) {
...
+                if ( env === 'all' ||  validBatch ) {
```
Comment (paetling) [inline reprocessFailedBatches.js]: think you really only need == here. can't have a different type that is also equal to 'all'.  But i guess it really does not matter

File: reprocessFailedBatches.js [hunk 4]
```diff
@@ -18,6 +19,14 @@ if (process.argv.length < 2) {
 }
 var setRegion = process.argv[2];
 
+if (process.argv.length < 3) {
+    console.log("You must prodive environment (prod, staging, all)");
+    process.exit(ERROR);
+}
+var env = process.argv[3];
+
+var dryRun = true;
```
Comment (paetling) [inline reprocessFailedBatches.js]: will this always be true?

**Comment by tleach** [comment]
Looks good i think

File: reprocessFailedBatches.js (hunk 4 above)
Comment (eriktaubeneck) [inline reprocessFailedBatches.js]: no, but also didn't want to over load with command line args. figured this was a good default and let people update it

**Pull request merged**